### PR TITLE
add adapter for YAML input

### LINKED ISSUE #22: YAML connector for spec
WIP: a version of https://github.com/metrumresearchgroup/mrgvalprep/blob/develop/R/read-spec-gsheet.R that takes a directory of YAML files

### PR DESCRIPTION
**Pull request by kyleam**
Support reading stories and requirements from YAML files.

Closes #22.

cc @kylebaron

### CODE REVIEW

**Review by seth127 — COMMENTED** [review]
I looked through the code but won't have time to run it today. All the test cases and spec, etc. look good to me. @kylebaron if you're waiting on this, you can feel free to pull it and test it on your stuff and approve the PR as you see fit. Otherwise, I should be able to get to it tomorrow or the next day.

File: inst/yaml-input/stories-only-1.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  description: >
+    S001 description
+  tests:
+  - TEST-ID-001
+  - TEST-ID-002
+  ProductRisk: low
+S002:
+  name: S002 name
+  description: >
+    S001 description
```
Comment (seth127) [inline inst/yaml-input/stories-only-1.yaml:12]: minor thing, but this should be `S002 description` (same with `S003` in other test file). This tripped me up when looking at these test results manually and seeing that whole column was the same.

File: R/read-spec-yaml.R [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+#'
+#' Those entries are used to construct a data frame with the following columns:
+#'
+#' * `StoryId` (character scalar)
+#' * `StoryName` (character scalar)
+#' * `StoryDescription` (character scalar)
+#' * `ProductRisk` (character scalar)
+#' * `TestIds` (character vector)
+#'
+#' @param stories,requirements Character vector of YAML files to read stories
```
Comment (seth127) [inline R/read-spec-yaml.R:107]: I had initially thought we would pass a dir here, but I guess this option is actually better. If you want everything in the dir you can easily to `stories = list.files(dir)`. Good call.

File: tests/testthat/test-read-spec-yaml.R [hunk 3]
```diff
@@ -0,0 +1,59 @@
+
+stories_only_cols <- c(
```
Comment (seth127) [inline tests/testthat/test-read-spec-yaml.R:2]: I would kind of prefer these go in `tests/testthat/setup.R` (and be all caps constants) just for consistency. Not a big deal if you have a reason to keep them here, but they seem potentially relevant to other tests not in this file.

Same with `yaml_file()` helper (move it to setup.R, not the all caps comment)

**Review by seth127 — CHANGES_REQUESTED** [review]
Mostly looks great. A few minor things on the test suite.

File: inst/yaml-input/stories-only-1.yaml (hunk 1 above)
Comment (kyleam) [reply inst/yaml-input/stories-only-1.yaml:12]: Thanks for spotting.  Will fix.

**Comment by kyleam** [comment]
Thanks for the review @seth127.  Updated.

**Review by seth127 — APPROVED** [review]
Looks good, thanks. Sorry to make your linter mad.

**Pull request merged**